### ISSUE TITLE
unexpected error: 'builtin.module' op region #0 ('bodyRegion') failed to verify constraint: region with 1 blocks

### ISSUE THREAD

**Issue by zhenxingsh** [llvm/llvm-project#62982]
When I try to convert `builtin.moduleOp` to `spirv.moduleOp`, such as:
from Intermediate dialect
```mlir
module attributes {
            ....
}
```
to spirv dialect
```mlir
spirv.module Logical GLSL450 requires #spirv.vce<v1.0, [Shader], []> {
           ...
}
```
the following coredump appears:
```console
**Stack dump:**
0.      Program arguments: ./build/bin/mlir-opt --verify-diagnostics -test-print-nesting --no-implicit-module -convert-relay-to-spirv -debug-only=dialect-conversion -mlir-print-op-generic -mlir-print-ir-module-scope -mlir-disable-threading --split-input-file ./utils/relay_mlir/flatten_relay.mlir -o spirv.mlir
 #0 0x000056084af583aa llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) ~/llvm-project/llvm/lib/Support/Unix/Signals.inc:602:22
 #1 0x000056084af58775 PrintStackTraceSignalHandler(void*) ~/llvm-project/llvm/lib/Support/Unix/Signals.inc:676:1
 #2 0x000056084af55f74 llvm::sys::RunSignalHandlers() ~/llvm-project/llvm/lib/Support/Signals.cpp:104:20
 #3 0x000056084af57cde SignalHandler(int) ~/llvm-project/llvm/lib/Support/Unix/Signals.inc:413:1
 #4 0x00007f4a22c63420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #5 0x00007f4a2270000b raise /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:51:1
 #6 0x00007f4a226df859 abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:81:7
 #7 0x00007f4a226df729 get_sysdep_segment_value /build/glibc-SzIz7B/glibc-2.31/intl/loadmsgcat.c:509:8
 #8 0x00007f4a226df729 _nl_load_domain /build/glibc-SzIz7B/glibc-2.31/intl/loadmsgcat.c:970:34
 #9 0x00007f4a226f0fd6 (/lib/x86_64-linux-gnu/libc.so.6+0x33fd6)
#10 0x000056084afaf40f llvm::ilist_iterator<llvm::ilist_detail::node_options<mlir::Operation, true, false, void>, false, false>::operator*() const ~/llvm-project/llvm/include/llvm/ADT/ilist_iterator.h:139:33
#11 0x000056084e847734 llvm::iplist_impl<llvm::simple_ilist<mlir::Operation>, llvm::ilist_traits<mlir::Operation>>::remove(llvm::ilist_iterator<llvm::ilist_detail::node_options<mlir::Operation, true, false, void>, false, false>&) ~/llvm-project/llvm/include/llvm/ADT/ilist.h:253:21
#12 0x000056084e90a937 llvm::iplist_impl<llvm::simple_ilist<mlir::Operation>, llvm::ilist_traits<mlir::Operation>>::erase(llvm::ilist_iterator<llvm::ilist_detail::node_options<mlir::Operation, true, false, void>, false, false>) ~/llvm-project/llvm/include/llvm/ADT/ilist.h:269:5
#13 0x000056084ea93adf llvm::iplist_impl<llvm::simple_ilist<mlir::Operation>, llvm::ilist_traits<mlir::Operation>>::erase(mlir::Operation*) ~/llvm-project/llvm/include/llvm/ADT/ilist.h:273:60
#14 0x000056084ea8ba73 mlir::Operation::erase() ~/llvm-project/mlir/lib/IR/Operation.cpp:432:1
#15 0x000056084e777d77 mlir::OwningOpRef<mlir::Operation*>::~OwningOpRef() ~/llvm-project/mlir/include/mlir/IR/OwningOpRef.h:39:3
#16 0x000056084e77539d performActions(llvm::raw_ostream&, std::shared_ptr<llvm::SourceMgr> const&, mlir::MLIRContext*, mlir::MlirOptMainConfig const&) ~/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:159:50
#17 0x000056084e7756ea processBuffer(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, mlir::MlirOptMainConfig const&, mlir::DialectRegistry&, llvm::ThreadPool*) ~/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:238:34
#18 0x000056084e7757bb mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, mlir::DialectRegistry&, mlir::MlirOptMainConfig const&)::'lambda'(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)::operator()(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&) const ~/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:268:36
#19 0x000056084e7766d4 mlir::LogicalResult llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>::callback_fn<mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, mlir::DialectRegistry&, mlir::MlirOptMainConfig const&)::'lambda'(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>(long, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&) ~/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:45:52
#20 0x000056084e8aa9ad llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>::operator()(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&) const ~/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:68:12
#21 0x000056084e8aa17f mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool)::'lambda'(llvm::StringRef)::operator()(llvm::StringRef) const ~/llvm-project/mlir/lib/Support/ToolUtilities.cpp:88:15
#22 0x000056084e8aa8f9 void llvm::interleave<llvm::StringRef const*, mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool)::'lambda'(llvm::StringRef), void llvm::interleave<llvm::SmallVector<llvm::StringRef, 8u>, mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool)::'lambda'(llvm::StringRef), llvm::raw_ostream, llvm::StringRef>(llvm::SmallVector<llvm::StringRef, 8u> const&, llvm::raw_ostream&, mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool)::'lambda'(llvm::StringRef), llvm::StringRef const&)::'lambda'(), void>(llvm::SmallVector<llvm::StringRef, 8u>, llvm::SmallVector<llvm::StringRef, 8u>, mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool)::'lambda'(llvm::StringRef), llvm::raw_ostream) ~/llvm-project/llvm/include/llvm/ADT/STLExtras.h:2110:3
#23 0x000056084e8aa8ad void llvm::interleave<llvm::SmallVector<llvm::StringRef, 8u>, mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool)::'lambda'(llvm::StringRef), llvm::raw_ostream, llvm::StringRef>(llvm::SmallVector<llvm::StringRef, 8u> const&, llvm::raw_ostream&, mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool)::'lambda'(llvm::StringRef), llvm::StringRef const&) ~/llvm-project/llvm/include/llvm/ADT/STLExtras.h:2131:13
#24 0x000056084e8aa7b9 mlir::splitAndProcessBuffer(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::function_ref<mlir::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>, llvm::raw_ostream&, bool, bool) ~/llvm-project/mlir/lib/Support/ToolUtilities.cpp:91:19
#25 0x000056084e775960 mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, mlir::DialectRegistry&, mlir::MlirOptMainConfig const&) ~/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:272:61
#26 0x000056084e7762c9 mlir::MlirOptMain(int, char**, llvm::StringRef, mlir::DialectRegistry&, bool) ~/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:367:13
#27 0x000056084aaf96a3 main ~/llvm-project/mlir/tools/mlir-opt/mlir-opt.cpp:255:32
#28 0x00007f4a226e1083 __libc_start_main /build/glibc-SzIz7B/glibc-2.31/csu/../csu/libc-start.c:342:3
#29 0x000056084aaf936e _start (./build/bin/mlir-opt+0x373636e)
```

**Is there a demo to implement `builtin.moduleOp` to `spirv.moduleOp`?**

**Comment by joker-eph** [comment]
This indicates a pretty bad bug in some of your passes, you should try to run with ASAN enabled.

**Comment by zhenxingsh** [comment]
> I guess the top level op matchAndRewrite operation is not currently supported in the entire module. lastly, solved it using another method during the output stage.
